### PR TITLE
H3 interop fixes

### PR DESCRIPTION
**Pull request by stammw**
Fixes needed to get better results at interop, along with some tooling and tracing.

### CODE REVIEW

**Comment by djc** [comment]
LGTM (pending test fixes).

**Comment by stammw** [comment]
I think this is ready now.

File: quinn-h3/src/connection.rs [hunk 1]
```diff
@@ -103,6 +103,7 @@ pub(crate) struct ConnectionInner {
 
 impl ConnectionInner {
     fn drive(&mut self, cx: &mut Context) -> Result<bool, DriverError> {
+        self.poll_recv_control(cx)?;
         self.poll_incoming_uni(cx)?;
         self.poll_send(cx)?;
         self.poll_recv_control(cx)?;
```
Comment (djc) [inline quinn-h3/src/connection.rs]: Should we poll this twice?

File: quinn-h3/examples/h3_server.rs [hunk 2]
```diff
@@ -62,12 +63,13 @@ async fn main() -> Result<()> {
         println!("server received connection");
         let connection = connecting
             .await
-            .map_err(|e| anyhow!("accept failed: {:?}", e))
-            .expect("server failed");
+            .map_err(|e| error!("accept failed: {:?}", e));
```
Comment (djc) [inline quinn-h3/examples/h3_server.rs]: Why do we no longer fail if the connection establishment fails? Perhaps we should at least log/print something in this case?

File: quinn-h3/src/proto/headers.rs [hunk 3]
```diff
@@ -273,13 +273,15 @@ impl Pseudo {
             },
         );
 
+        let len = if authority.is_some() { 1 } else { 0 };
```
Comment (djc) [inline quinn-h3/src/proto/headers.rs]: Maybe just put the `3 + ` in here?

File: quinn-h3/src/proto/connection.rs [hunk 4]
```diff
@@ -329,7 +329,13 @@ mod tests {
     fn decode_header() {
         let mut header_map = HeaderMap::new();
         header_map.append("hello", HeaderValue::from_static("text/html"));
-        let header = Header::request(Method::GET, Uri::default(), header_map);
+        let header = Header::request(
+            Method::GET,
+            // NOTE: H3 adds a trailing `/`, so the one in following the url is important
+            //       only to make the `Header` comparaison succeed at the end of this test.
```
Comment (djc) [inline quinn-h3/src/proto/connection.rs]: Typo in `comparaison`

File: quinn-h3/examples/h3_server.rs (hunk 2 above)
Comment (stammw) [reply quinn-h3/examples/h3_server.rs]: We want to continue accepting the following incoming connections, so we don't want to fail `main()`. 

I guess there is something logged with the `error!()` call. 

Writing this with `if let` is a bit convoluted. I'll make this a `match`, will read much better.

Comment (djc) [reply quinn-h3/examples/h3_server.rs]: Ah, if `error!()` logs then we should be fine.

File: quinn-h3/src/proto/connection.rs (hunk 4 above)
Comment (stammw) [reply quinn-h3/src/proto/connection.rs]: Fixed.

File: quinn-h3/src/proto/headers.rs (hunk 3 above)
Comment (stammw) [reply quinn-h3/src/proto/headers.rs]: oh yeah, thanks

File: quinn-h3/src/connection.rs (hunk 1 above)
Comment (stammw) [reply quinn-h3/src/connection.rs]: Got it better in a separate commit. Polling control upon opening. It's important to use the settings (first frame on control stream) as soon as possible as it contains compression enablement parameters.

File: quinn-h3/examples/h3_server.rs (hunk 2 above)
Comment (stammw) [reply quinn-h3/examples/h3_server.rs]: Rewrote it using `match`

**Pull request merged**